lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py

```diff
@@ -0,0 +1,379 @@
+        required: True
+    directory:
+        description:
+            - Create or delete directory
+            - 'Valid attributes are:'
+            - '  path: directory path to create or remove'
+            - '  operation: Valid values are create, delete'
+            - '  recurse (boolean): Not required, default (false)'
+        required: False
+    file_copy:
+        description:
+            - Copy file to vm networkless
+            - 'Valid attributes are:'
+            - '  src: file source absolute or relative'
+            - '  dest: file destination, path must be exist'
+            - '  overwrite: False or True (not required, default False)'
+        required: False
+    fetch_file:
+        description:
+            - Get file from vm networkless
+            - 'Valid attributes are:'
+            - '  src: The file on the remote system to fetch. This I(must) be a file, not a directory'
+            - '  dest: file destination on localhost, path must be exist'
+        required: False
+
+'''
+
+EXAMPLES = '''
+- name: Create directory inside a vm
+  vmware_guest_file_operation:
+    hostname: myVSphere
+    username: myUsername
+    password: mySecret
+    datacenter: myDatacenter
+    validate_certs: True
+    folder: /vm
+    vm_id: NameOfVM
+    vm_username: root
+    vm_password: superSecret
+    directory:
+      path: "/test"
+      operation: create
+      recurse: no
+  delegate_to: localhost
+
+- name: copy file to vm
+  vmware_guest_file_operation:
+    hostname: myVSphere
+    username: myUsername
+    password: mySecret
+    datacenter: myDatacenter
+    validate_certs: True
+    folder: /vm
+    vm_id: NameOfVM
+    vm_username: root
+    vm_password: superSecret
+    copy:
+        src: "files/test.zip"
+        dest: "/root/test.zip"
+        overwrite: False
+  delegate_to: localhost
+
+- name: fetch file from vm
+  vmware_guest_file_operation:
+    hostname: myVSphere
+    username: myUsername
+    password: mySecret
+    datacenter: myDatacenter
+    validate_certs: True
+    folder: /vm
+    vm_id: NameOfVM
+    vm_username: root
+    vm_password: superSecret
+    fetch:
+        src: "/root/test.zip"
+        dest: "files/test.zip"
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils import urls
+from ansible.module_utils._text import to_bytes, to_native
+from ansible.module_utils.vmware import (connect_to_api, find_cluster_by_name, find_datacenter_by_name,
+                                         find_vm_by_id, HAS_PYVMOMI, vmware_argument_spec, PyVmomi)
+
+
+def directory(module, content, vm):
+    operations_permit = ['create', 'delete']
+    vm_username = module.params['vm_username']
+    vm_password = module.params['vm_password']
+
+    if "path" not in module.params["directory"]:
```

Comment:
I prefer to work with sub options, examples: 
https://github.com/ansible/ansible/blob/5e6b2495c045961b73a11a31b71b971c16bea9c7/lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup.py#L337-L364
https://github.com/ansible/ansible/blob/5e6b2495c045961b73a11a31b71b971c16bea9c7/lib/ansible/modules/network/nuage/nuage_vspk.py#L985-L1026

The last one also shows how you can define requirements on combinations